stan/math/fwd/functor/fvar_finite_diff.hpp

```diff
@@ -0,0 +1,113 @@
+#ifndef STAN_MATH_FWD_FUNCTOR_FVAR_FINITE_DIFF_HPP
+#define STAN_MATH_FWD_FUNCTOR_FVAR_FINITE_DIFF_HPP
+
+#include <stan/math/prim/meta.hpp>
+#include <stan/math/prim/functor/apply_scalar_binary.hpp>
+#include <stan/math/prim/functor/finite_diff_gradient_auto.hpp>
+#include <stan/math/prim/fun/value_of.hpp>
+#include <stan/math/prim/fun/sum.hpp>
+#include <stan/math/prim/fun/serializer.hpp>
+
+namespace stan {
+namespace math {
+namespace internal {
+/**
+ * Helper function for aggregating tangents if the respective input argument
+ * was an fvar<T> type.
+ *
+ * Overload for when the input is not an fvar<T> and no tangents are needed.
+ *
+ * @tparam FuncTangentT Type of tangent calculated by finite-differences
+ * @tparam InputArgT Type of the function input argument
+ * @param tangent Calculated tangent
+ * @param arg Input argument
+ */
+template <typename FuncTangentT, typename InputArgT,
+          require_not_st_fvar<InputArgT>* = nullptr>
+inline constexpr double aggregate_tangent(const FuncTangentT& tangent,
+                                          const InputArgT& arg) {
+  return 0;
+}
+
+/**
+ * Helper function for aggregating tangents if the respective input argument
+ * was an fvar<T> type.
+ *
+ * Overload for when the input is an fvar<T> and its tangent needs to be
+ * aggregated.
+ *
+ * @tparam FuncTangentT Type of tangent calculated by finite-differences
+ * @tparam InputArgT Type of the function input argument
+ * @param tangent Calculated tangent
+ * @param arg Input argument
+ */
+template <typename FuncTangentT, typename InputArgT,
+          require_st_fvar<InputArgT>* = nullptr>
+auto aggregate_tangent(const FuncTangentT& tangent, const InputArgT& arg) {
+  return sum(apply_scalar_binary(
+      tangent, arg, [](const auto& x, const auto& y) { return x * y.d_; }));
+}
+}  // namespace internal
+
+/**
+ * This frameworks adds fvar<T> support for arbitrary functions through
+ * finite-differencing. Higher-order inputs (i.e., fvar<var> & fvar<fvar<T>>)
+ * are also implicitly supported.
+ *
+ * @tparam F Type of functor for which fvar<T> support is needed
+ * @tparam TArgs... Types of arguments (containing at least one fvar<T> type)
+ *                     to be passed to function
+ * @param func Functor for which fvar<T> support is needed
+ * @param args Parameter pack of arguments to be passed to functor.
+ */
+template <typename F, typename... TArgs,
+          require_any_st_fvar<TArgs...>* = nullptr>
+auto fvar_finite_diff(const F& func, const TArgs&... args) {
+  using FvarT = return_type_t<TArgs...>;
+  using FvarInnerT = typename FvarT::Scalar;
+
+  auto serialised_args = serialize<FvarInnerT>(value_of(args)...);
+
+  // Create a 'wrapper' functor which will take the flattened column-vector
+  // and transform it to individual arguments which are passed to the
+  // user-provided functor
+  auto serial_functor
+      = [&](const auto& v) { return func(to_deserializer(v).read(args)...); };
```

Comment:
`to_deserializer(v).read(args)...` is going to make a `deserializer` for every `arg` right? Or maybe I'm misreading the expansion?

```suggestion
      = [&](const auto& v) { 
          auto deserial = to_deserializer(v);
          return func(deserial.read(args)...); 
        };
```

Reply:
Ahh good catch, you're right! Fixed up here and in your other comment